### DISCUSSION TITLE
Chrome iOS + auto-generated USDZ = not launching Quick-Look?

### DISCUSSION THREAD

**Discussion by cunicode** [google/model-viewer#2898] [Q&A]
Chrome iOS fails to launch Quick-look when the USDZ is auto-generated by modelviewer. 

can you reproduce? or is it only me? 

√ - The auto-generated USDZ loads in Quick-Look in Safari 
√ - Chrome iOS launches Quick-look if if USDZ is in the code 
X - Chrome iOS does not launch if the USDZ is auto-generated

**Comment by samaneh-kazemi** [comment]
@cunicode Hi, can you please share a repro link so we can debug your case?

**Reply by cunicode** [reply]
this one: https://modelviewer.dev/examples/augmentedreality/#customButton 

```
<!-- Import the component -->
<model-viewer ar ar-modes="webxr scene-viewer quick-look" camera-controls src="../../shared-assets/models/Astronaut.glb" alt="A 3D model of an astronaut">
  <button slot="ar-button" style="background-color: white; border-radius: 4px; border: none; position: absolute; top: 16px; right: 16px; ">
      👋 Activate AR
  </button>
</model-viewer>
```

**Reply by elalish** [reply]
I believe this has been fixed by the latest iOS Chrome version. Have you updated?

**Comment by milesgreen** [comment]
I don’t think Chrome iOS supports opening USDZ from auto-generated file blob, only from URL.

To work in Chrome iOS, you need to provide a path to a USDZ in the ios-src parameter. 

Which is a pain, as that limits USDZ auto generation to just Safari on iOS.

**Comment by cunicode** [comment]
that's the behaviour I was seeing. 
I just wanted to confirm if that was the expected behaviour. 
if yes, documentation should be updated.

**Reply by elalish** [reply]
It's just a  bug on iOS Chrome; they're  working on it. Similar to how Safari has a bug where Quick  Look's API doesn't work for auto-generated USDZ either, since they drop query  params  on blobs. Ah for life  at the bleeding edge...

**Reply by nbaybara** [reply]
Is there any updates?

**Comment by uprealdigital** [comment]
Hi, 

I just ran into that issue with IOS Chrome and auto-generated USDZ files.

I saw a GLB / GLTF exporter but if we had a USDZ exporter, we could generate the file ourselves and serve the file in a regular URL and not a blob URL right?

**Reply by elalish** [reply]
That's right; you can use Apple's converter for this.

**Comment by nbaybara** [comment]
Is there any updates on this issue?

**Comment by rawwerks** [comment]
i am also curious if there is any update on this issue? 

related discussions: 

- https://github.com/google/model-viewer/discussions/3192
- https://github.com/google/model-viewer/discussions/3184

**Comment by elalish** [comment]
The main thread is here: https://github.com/google/model-viewer/issues/3292 which includes a link to the iOS Chrome bug, which you could +1: https://bugs.chromium.org/p/chromium/issues/detail?id=1197345
